keps/sig-auth/4872-harden-kubelet-cert-validation/README.md

```diff
@@ -0,0 +1,389 @@
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Other
+  - Details: users can create a Node with a kubelet serving certificate that doesn't meet the CN requirements enforced by this validation (something different than `system:node:<node-name>`).Then run `kubectl logs` for any pod running in that node. If it returns an error for an invalid certificate, the feature is working.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+The average `apiserver_request_duration_seconds` for logs/exec/port-forward requests is within reasonable limits.
+A raising value after enabling this feature could signal overhead introduced by the extra validation.
+
+> TODO: I expect the overhead to be negligible and probably to fall in within the standard deviation of the current average. Specially for long running requests like port-forward and exec. Is this even valuable to have here?
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - Metric name: `kube_apiserver_pod_logs_backend_tls_failure_total`
+  - Components exposing the metric: kube-apiserver
+
+> TODO: should `kube_apiserver_pod_logs_backend_tls_failure_total` reflect errors due to the new CN validation?
+> It's technically a TLS failure, but it's not part of the base TLS client validations.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+We could add a metric specific to track the number of requests that failed due to the new CN validation. In addition, we could track the time spent per request on the CN validation.
+
+However, we consider these metrics to not provide enough value to justify the work to maintain them.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No. This only affects streaming APIs and these are not covered by SLIs/SLOs.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+Note: depending on the implementation (caching the client-go transport or not) there might be a slight increase in memory (due to one transport per node being cached) or in CPU usage (due to building the transport on the fly for every request). This should be negligible.
```

Comment:
The implementation will be required not to create extra TLS connections from API server to nodes.